# Awesome-metagenomic
awesome softwares, pipelines and resources for metagenomic

## toolkits

* [MOCAT2](http://vm-lux.embl.de/~kultima/MOCAT/): Process & analyze metagenomes

## Database

* [Resfams](http://www.dantaslab.org/resfams/): a curated database of protein families and associated profile hidden Markov models (HMMs), confirmed for antibiotic resistance function and organized by ontology.

## Binning

* [BinSanity](https://github.com/edgraham/BinSanity): Unsupervised Clustering of Environmental Microbial Assemblies Using Coverage and Affinity Propagation
* [CONCOCT](https://github.com/BinPro/CONCOCT): Clustering cONtigs with COverage and ComposiTion (https://concoct.readthedocs.org/)
* [Assigning Whole Metagenome Taxonomy](https://www.protocols.io/view/Script-P7-Assigning-Whole-Metagenome-Taxonomy-egubbww)
* [GroopM](https://github.com/Ecogenomics/GroopM): Metagenomic binning suite (https://ecogenomics.github.io/GroopM/)
* [MyCC](http://sourceforge.net/projects/sb2nhri/files/MyCC/): Accurate binning of metagenomic contigs via automated clustering sequences using information of genomic signatures and marker genes (https://www.nature.com/articles/srep24175)
* [MetaBat](https://bitbucket.org/berkeleylab/metabat): an efficient tool for accurately reconstructing single genomes from complex microbial communities
* [MaxBin2](https://sourceforge.net/projects/maxbin2/): an automated binning algorithm to recover genomes from multiple metagenomic datasets (https://academic.oup.com/bioinformatics/article/32/4/605/1744462)
* [DAS Tool](https://github.com/cmks/DAS_Tool): automated method that integrates the results of a flexible number of binning algorithms to calculate an optimized, non-redundant set of bins from a single assembly
* [COCACOLA](https://github.com/younglululu/COCACOLA): binning metagenomic contigs using sequence COmposition, read CoverAge, CO-alignment and paired-end read LinkAge



## Taxonomy

* [metametamerge](https://github.com/pirovc/metametamerge)
* [MetaPalette](https://github.com/dkoslicki/MetaPalette): Metagenomic profiling and phylogenetic distances via common kmers (https://github.com/pirovc/metameta/)
* [mOTU](http://www.bork.embl.de/software/mOTU/): Taxonomic profiling using mOTUs
* [MetaMeta](https://gitlab.com/rki_bioinformatics): integrating metagenome analysis tools to improve taxonomic profiling (https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0318-y)
* [kaiju](https://github.com/bioinformatics-centre/kaiju): Fast taxonomic classification of metagenomic sequencing reads using a protein reference database
* [LMAT](http://sourceforge.net/projects/lmat): Scalable metagenomic taxonomy classification using a reference genome database (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3753567/)
* [Clark](http://clark.cs.ucr.edu/Overview/): Fast, accurate and versatile sequence classification system
* [metaphlan2](https://bitbucket.org/biobakery/biobakery/wiki/metaphlan2): a tool for profiling the composition of microbial communities from metagenomic shotgun sequencing data.
* [krakenhll](https://github.com/fbreitwieser/krakenhll): Metagenomics classifier with unique k-mer counting for more specific results
* [Kraken](http://ccb.jhu.edu/software/kraken/MANUAL.html): taxonomic sequence classification system
* [MetaOthello](https://github.com/xa6xa6/metaOthello): A novel data structure to support ultra-fast taxonomic classification of metagenomic sequences with k-mer signatures
* [k-SLAM](https://github.com/aindj/k-SLAM): accurate and ultra-fast taxonomic classification and gene identification for large metagenomic data sets 
* [Phylosift](https://phylosift.wordpress.com): mining the global metagenome




## Visualization

* [Anvi’o](): an advanced analysis and visualization platform for ‘omics data
* [Pavian](https://github.com/fbreitwieser/pavian):  Interactive analysis of metagenomics data for microbiomics and pathogen identification


## Function

* [STAMP](http://kiwi.cs.dal.ca/Software/STAMP) : oftware package for analyzing taxonomic or metabolic profiles that promotes ‘best practices’ in choosing appropriate statistical techniques and reporting results. 




## Protocols

* [Binning Procedure applied to Tara Oceans Dataset using BinSanity](https://www.protocols.io/view/binning-procedure-applied-to-tara-oceans-dataset-u-iwgcfbw)
* [Pre-Processing Samples](https://www.protocols.io/view/Script-P1-Pre-Processing-Samples-edrba56)
* [Contig Assembly](https://www.protocols.io/view/Script-P2-Contig-Assembly-ed9ba96)
* [Open Reading Frame Prediction](https://www.protocols.io/view/Script-P3-Open-Reading-Frame-Prediction-efpbbmn)
* [Virome Alpha Diversity](https://www.protocols.io/view/Script-P5-Diversity-efrbbm6)
[Assigning Viral Taxonomy](https://www.protocols.io/view/Script-P4-Assigning-Viral-Taxonomy-efqbbmw)
* [HPyV Detection](https://www.protocols.io/view/Script-P6-HPyV-Detection-efxbbpn)
[Assigning Whole Metagenome Taxonomy](https://www.protocols.io/view/Script-P7-Assigning-Whole-Metagenome-Taxonomy-egubbww)
* [Phage Replication Cycle](https://www.protocols.io/view/Script-P8-Phage-Replication-Cycle-egwbbxe)
* [Phage-Bacteria Interaction Network](https://www.protocols.io/view/Script-P9-Phage-Bacteria-Interaction-Network-egybbxw)
* [CRISPR Identification](https://www.protocols.io/view/Script-P10-CRISPR-Identification-ehdbb26)
* [Functional Analysis](https://www.protocols.io/view/Script-P11-Functional-Analysis-ehubb6w)
* [Antibiotic Resistance](https://www.protocols.io/view/Script-P12-Antibiotic-Resistance-ehwbb7e)
* [An anvi'o tutorial using the Infant Gut Dataset](http://merenlab.org/tutorials/infant-gut/)




## Protocols for plotting

* [Virome Contig and Sequencing Statistics](https://www.protocols.io/view/Script-R1-Virome-Contig-and-Sequencing-Statistics-eh5bb86)
* [Whole Metagenome Contig and Sequencing Statistics](https://www.protocols.io/view/Script-R2-Whole-Metagenome-Contig-and-Sequencing-S-eh6bb9e)
* [Quality Control Statistics](https://www.protocols.io/view/Script-R3-Quality-Control-Statistics-eh9bb96)
* [Virome Taxonomy](https://www.protocols.io/view/Script-R4-Virome-Taxonomy-eiabcae)
* [Virome Alpha Diversity](https://www.protocols.io/view/Script-R5-Virome-Alpha-Diversity-eimbcc6)
* [Virome Beta Diversity](https://www.protocols.io/view/Script-R6-Virome-Beta-Diversity-einbcde)
* [Intra and Inter Personal Dissimilarity](https://www.protocols.io/view/Script-R7-Intra-and-Inter-Personal-Dissimilarity-eipbcdn)
* [Plotting Bacterial Taxonomy from MetaPhlan](https://www.protocols.io/view/Script-R8-Plotting-Bacterial-Taxonomy-from-MetaPhl-ejbbcin)
* [Plotting Microbial Taxonomy from MEGAN](https://www.protocols.io/view/Script-R9-Plotting-Microbial-Taxonomy-from-MEGAN-ejdbci6)
* [Whole Metagenome Beta Diversity](https://www.protocols.io/view/Script-R10-Whole-Metagenome-Beta-Diversity-ejebcje)
* [Replication Cycle](https://www.protocols.io/view/Script-R11-Replication-Cycle-ejfbcjn)
* [Functional Analysis](https://www.protocols.io/view/Script-R12-Functional-Analysis-ejibcke)
* [OPF and AMG Analysis](https://www.protocols.io/view/Script-R13-OPF-and-AMG-Analysis-ejkbckw)
* [CARD Antibiotic Resistance Analysis](https://www.protocols.io/view/Script-R14-CARD-Antibiotic-Resistance-Analysis-ejnbcme)
* [CoNet Analysis - Formatting Relative Abundance Files](https://www.protocols.io/view/Script-R15-CoNet-Analysis-Formatting-Relative-Abun-ejpbcmn)
* [CRISPR Analysis](https://www.protocols.io/view/Script-R16-CRISPR-Analysis-ejqbcmw)



## Websites

* [TARA OCEANS](http://ocean-microbiome.embl.de/companion.html): Structure and function of the global ocean microbiome

