# Awesome-metagenomic
awesome softwares, pipelines and resources for metagenomic


## Binning

* [BinSanity](https://github.com/edgraham/BinSanity): Unsupervised Clustering of Environmental Microbial Assemblies Using Coverage and Affinity Propagation
* [CONCOCT]()
* [MetaBat]()
* [MyCC]()
* [MyCC]()
* [GroopM]()

## Visualization

* [Anvi’o](): an advanced analysis and visualization platform for ‘omics data

## Protocols

* [Binning Procedure applied to Tara Oceans Dataset using BinSanity](https://www.protocols.io/view/binning-procedure-applied-to-tara-oceans-dataset-u-iwgcfbw)
* [Pre-Processing Samples](https://www.protocols.io/view/Script-P1-Pre-Processing-Samples-edrba56)
* [Contig Assembly](https://www.protocols.io/view/Script-P2-Contig-Assembly-ed9ba96)
* [Open Reading Frame Prediction](https://www.protocols.io/view/Script-P3-Open-Reading-Frame-Prediction-efpbbmn)
* [Virome Alpha Diversity](https://www.protocols.io/view/Script-P5-Diversity-efrbbm6)
[Assigning Viral Taxonomy](https://www.protocols.io/view/Script-P4-Assigning-Viral-Taxonomy-efqbbmw)
* [HPyV Detection](https://www.protocols.io/view/Script-P6-HPyV-Detection-efxbbpn)
[Assigning Whole Metagenome Taxonomy](https://www.protocols.io/view/Script-P7-Assigning-Whole-Metagenome-Taxonomy-egubbww)
* [Phage Replication Cycle](https://www.protocols.io/view/Script-P8-Phage-Replication-Cycle-egwbbxe)
* [Phage-Bacteria Interaction Network](https://www.protocols.io/view/Script-P9-Phage-Bacteria-Interaction-Network-egybbxw)
* [CRISPR Identification](https://www.protocols.io/view/Script-P10-CRISPR-Identification-ehdbb26)
* [Functional Analysis](https://www.protocols.io/view/Script-P11-Functional-Analysis-ehubb6w)
* [Antibiotic Resistance](https://www.protocols.io/view/Script-P12-Antibiotic-Resistance-ehwbb7e)


## Protocols for plotting

* [Virome Contig and Sequencing Statistics](https://www.protocols.io/view/Script-R1-Virome-Contig-and-Sequencing-Statistics-eh5bb86)
* [Whole Metagenome Contig and Sequencing Statistics](https://www.protocols.io/view/Script-R2-Whole-Metagenome-Contig-and-Sequencing-S-eh6bb9e)
* [Quality Control Statistics](https://www.protocols.io/view/Script-R3-Quality-Control-Statistics-eh9bb96)
* [Virome Taxonomy](https://www.protocols.io/view/Script-R4-Virome-Taxonomy-eiabcae)
* [Virome Alpha Diversity](https://www.protocols.io/view/Script-R5-Virome-Alpha-Diversity-eimbcc6)
* [Virome Beta Diversity](https://www.protocols.io/view/Script-R6-Virome-Beta-Diversity-einbcde)
* [Intra and Inter Personal Dissimilarity](https://www.protocols.io/view/Script-R7-Intra-and-Inter-Personal-Dissimilarity-eipbcdn)
* [Plotting Bacterial Taxonomy from MetaPhlan](https://www.protocols.io/view/Script-R8-Plotting-Bacterial-Taxonomy-from-MetaPhl-ejbbcin)
* [Plotting Microbial Taxonomy from MEGAN](https://www.protocols.io/view/Script-R9-Plotting-Microbial-Taxonomy-from-MEGAN-ejdbci6)
* [Whole Metagenome Beta Diversity](https://www.protocols.io/view/Script-R10-Whole-Metagenome-Beta-Diversity-ejebcje)
* [Replication Cycle](https://www.protocols.io/view/Script-R11-Replication-Cycle-ejfbcjn)
* [Functional Analysis](https://www.protocols.io/view/Script-R12-Functional-Analysis-ejibcke)
* [OPF and AMG Analysis](https://www.protocols.io/view/Script-R13-OPF-and-AMG-Analysis-ejkbckw)
* [CARD Antibiotic Resistance Analysis](https://www.protocols.io/view/Script-R14-CARD-Antibiotic-Resistance-Analysis-ejnbcme)
* [CoNet Analysis - Formatting Relative Abundance Files](https://www.protocols.io/view/Script-R15-CoNet-Analysis-Formatting-Relative-Abun-ejpbcmn)
* [CRISPR Analysis](https://www.protocols.io/view/Script-R16-CRISPR-Analysis-ejqbcmw)




