## Nanopore tools

### Quality control

* [scrappie basecaller](https://github.com/nanoporetech/scrappie): Scrappie is a technology demonstrator for the Oxford Nanopore Research Algorithms group

* [Porechop](https://github.com/rrwick/Porechop): adapter trimmer for Oxford Nanopore reads
```
$ conda install -c bioconda porechop
```

* [picopore](https://github.com/scottgigante/picopore): A tool to reduce the size of Oxford Nanopore Technologies' datasets without losing information
```
$ conda install -c bioconda picopore
```

* [Filtlong](https://github.com/rrwick/Filtlong): quality filtering tool for long reads

* [Basecalling-comparison](https://github.com/rrwick/Basecalling-comparison): A comparison of different Oxford Nanopore basecallers

* [nanopolish](https://github.com/jts/nanopolish): Signal-level algorithms for MinION data
```
$ conda install -c bioconda nanopolish 
```

* [Chiron](https://github.com/haotianteng/Chiron): A basecaller for Oxford Nanopore Technologies' sequencers
```
$ git clone https://github.com/haotianteng/chiron.git
```




### Assembly

* [Unicycler ](https://github.com/rrwick/Unicycler): hybrid assembly pipeline for bacterial genomes

* [ont-assembly-polish](https://github.com/nanoporetech/ont-assembly-polish): ONT assembly and Illumina polishing pipeline

* [Bandage](https://github.com/rrwick/Bandage): a Bioinformatics Application for Navigating De novo Assembly Graphs Easily

### Manuals

* [nanopore-scripts](https://github.com/scottgigante/nanopore-scripts): Various scripts and utilities for manipulation of nanopore data

* [Bacterial-genome-assemblies-with-multiplex-MinION-sequencing](https://github.com/rrwick/Bacterial-genome-assemblies-with-multiplex-MinION-sequencing)


